frontend/src/metabase/lib/time.ts

```diff
@@ -197,3 +197,45 @@ export function parseTimestamp(
   }
   return isLocal ? m.local() : m;
 }
+
+const isTimezoneInOffsetFormat = (timezone: string) => {
+  const offsetPattern = /^[+-]\d{2}:\d{2}$/;
+  return offsetPattern.test(timezone);
+};
+
+const getTimezoneWithOffset = (offsetString: string) => {
+  const offsetSign = offsetString[0];
+  const offsetHours = parseInt(offsetString.substring(1, 3));
+  const offsetMinutes = parseInt(offsetString.substring(4, 6));
+
+  const totalOffsetMinutes =
+    (offsetHours * 60 + offsetMinutes) * (offsetSign === "+" ? 1 : -1);
+
+  const matchingZones = moment.tz.names().filter(zone => {
+    const tzOffset = moment.tz(zone).utcOffset();
+    return tzOffset === totalOffsetMinutes;
+  });
+
+  const etcZone = matchingZones.find(zone => zone.startsWith("Etc/GMT"));
+  return etcZone || matchingZones[0];
+};
+
+export const getIanaTimezone = (timezone: string) => {
+  if (!isTimezoneInOffsetFormat(timezone)) {
+    return timezone;
+  }
+
+  try {
+    const matchingTimezone = getTimezoneWithOffset(timezone);
+
+    if (!matchingTimezone) {
+      console.warn(`Could not find a time zone with offset ${timezone}`);
+      return timezone;
+    }
+
+    return matchingTimezone;
+  } catch (e) {
+    console.warn(e);
```

Comment:
should we also add a prefix message so we knows where the message comes from, when the sourcemap isn't available?

Reply:
I removed this code as I decided to go with a different approach